src/dmd/target.d

```diff
@@ -839,5 +843,20 @@ struct TargetCPP
     }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/**
+ * Functions and variables specific to interface with extern(Objective-C) ABI.
+ */
+struct TargetObjC
+{
+    bool supported;     /// set if compiler can inteface with Objective-C
+
+    extern (D) void initialize(ref const Param params, ref const Target target)
```

Comment:
Your assuming that it will be D code that sets this value.

Reply:
I'm not sure if it should be an enum, as there is more than one objective-c abi.  Probably shouldn't think about that unless I actually look into it (which will likely be never given my free time).

Reply:
> You're assuming that it will be D code that sets this value.

Yes, because `initialize` is marked as `extern(D)`. Or do you plan to set the field directly?

Reply:
> I'm not sure if it should be an enum, as there is more than one objective-c abi. Probably shouldn't think about that unless I actually look into it (which will likely be never given my free time).

There are three or four different ABIs, depending on how your counting [1]:

* Apple Modern ABI (x86-64, ARM, ARM64)
* Apple Legacy ABI (PPC, PPC64, x86)
    * With support for Objective-C 2.0 features
    * Without support for Objective-C 2.0 features
* GNU ABI (other platforms)

I wouldn't bother with supporting anything other than the Apple Modern ABI.

[1] https://gcc.gnu.org/onlinedocs/gcc/Objective-C-and-Objective-C_002b_002b-Dialect-Options.html